### ISSUE TITLE
Make dart booster part of gbtree

### ISSUE THREAD

**Issue by RAMitchell** [dmlc/xgboost#12031]
We are deprecating "booster=gblinear" (#12023 ) , that leaves only "gbtree" and "dart". Internally dart is really just a thin skin over gbtree that modifies tree weights. We can instead make dart just a parameter of gbtree.

This reduces complexity around GBTree/Learner and supports future refactoring.

There are risks however around backwards compatibility and breaking changes, so care is required.

**Comment by RAMitchell** [comment]
Now that I am looking at the documentation for dart I am wondering why dropout is applied for non-training predict automatically? We know when we are training or not so this is an unusual default.

**Comment by trivialfis** [comment]
Can we close this along with https://github.com/dmlc/xgboost/pull/12032 ?
